analyzers/src/SonarAnalyzer.CSharp/Rules/ReturnEmptyCollectionInsteadOfNull.cs

```diff
@@ -18,68 +18,70 @@
+        context.RegisterNodeAction(ReportIfReturnsNullOrDefault,
+            SyntaxKind.MethodDeclaration,
+            SyntaxKindEx.LocalFunctionStatement,
+            SyntaxKind.PropertyDeclaration,
+            SyntaxKind.OperatorDeclaration);
 
-        private static void ReportIfReturnsNullOrDefault(SonarSyntaxNodeReportingContext context)
+    private static void ReportIfReturnsNullOrDefault(SonarSyntaxNodeReportingContext context)
+    {
+        if (GetExpressionBody(context.Node) is { } expressionBody)
         {
-            if (GetExpressionBody(context.Node) is { } expressionBody)
-            {
-                var nullOrDefaultLiterals = GetNullOrDefaultExpressions(expressionBody.Expression)
-                    .Select(statement => statement.GetLocation())
-                    .ToList();
+            var nullOrDefaultLiterals = GetNullOrDefaultExpressions(expressionBody.Expression)
+                .Select(statement => statement.GetLocation())
+                .ToList();
 
-                ReportIfAny(nullOrDefaultLiterals);
-            }
-            else if (GetBody(context.Node) is { } body)
-            {
-                var nullOrDefaultLiterals = GetReturnNullOrDefaultExpressions(body)
-                    .Select(returnStatement => returnStatement.GetLocation())
-                    .ToList();
+            ReportIfAny(nullOrDefaultLiterals);
+        }
+        else if (GetBody(context.Node) is { } body)
+        {
+            var nullOrDefaultLiterals = GetReturnNullOrDefaultExpressions(body)
+                .Select(returnStatement => returnStatement.GetLocation())
+                .ToList();
 
-                ReportIfAny(nullOrDefaultLiterals);
-            }
+            ReportIfAny(nullOrDefaultLiterals);
+        }
 
-            void ReportIfAny(List<Location> nullOrDefaultLiterals)
+        void ReportIfAny(List<Location> nullOrDefaultLiterals)
+        {
+            if (nullOrDefaultLiterals.Count > 0 && IsReturningCollection(context))
             {
-                if (nullOrDefaultLiterals.Count > 0 && IsReturningCollection(context))
-                {
-                    context.ReportIssue(Rule.CreateDiagnostic(context.Compilation, nullOrDefaultLiterals[0], additionalLocations: nullOrDefaultLiterals.Skip(1)));
-                }
+                context.ReportIssue(Rule.CreateDiagnostic(context.Compilation, nullOrDefaultLiterals[0], additionalLocations: nullOrDefaultLiterals.Skip(1)));
             }
         }
+    }
 
-        private static bool IsReturningCollection(SonarSyntaxNodeReportingContext context)
-        {
-            var symbol = context.SemanticModel.GetDeclaredSymbol(context.Node);
-            var methodSymbol = (symbol as IPropertySymbol)?.GetMethod ?? symbol as IMethodSymbol;
+    private static bool IsReturningCollection(SonarSyntaxNodeReportingContext context) =>
+        GetReturnMethod(context) is { } method
```

Comment:
`ReturnType` can be extracted directly
```suggestion
        GetReturnMethod(context) is { ReturnType: { } returnType }
```